server/src/main/java/io/littlehorse/common/util/RocksConfigSetter.java

```diff
@@ -1,39 +1,89 @@
 package io.littlehorse.common.util;
 
+import io.littlehorse.common.LHServerConfig;
 import java.util.Map;
 import lombok.extern.slf4j.Slf4j;
 import org.apache.kafka.streams.state.RocksDBConfigSetter;
+import org.apache.kafka.streams.state.internals.BlockBasedTableConfigWithAccessibleCache;
+import org.rocksdb.Cache;
 import org.rocksdb.CompactionStyle;
 import org.rocksdb.Options;
 
 @Slf4j
 public class RocksConfigSetter implements RocksDBConfigSetter {
 
+    // Need to inject the LHServerConfig, but Kafka Streams requires we pass in a Class with
+    // a default constructor. So the only way to do that is to have a static singleton which
+    // we set elsewhere in the code.
+    public static LHServerConfig serverConfig;
```

Comment:
@mjsax this appears to be the only way to "inject" configuration into the `RockDBConfigSetter` implementation. Would it be worth a KIP to add a method:

```
KafkaStreams#setRocksDBConfigurator(RocksDBConfigSetter otterSetter)
```
The method should throw `IllegalStateException` if it is called after `KafkaStreams#start()`, or if the `rocksdb.config.setter` parameter is set on the properties.

Reply:
Don't think it's the only way to inject configs.

There is `RocksConfigSetter.setConfig(..., config)` taking the Kafka Streams `config` object and third parameter. When you create `StreamsConfig` you pass in `Map<?, ?> props` and all entries will be contained in `config`.

Thus, you can take `LHServerConfig` and add all of it to `props` before creating `StreamsConfig` and should have access to all of it in `setConfig(...)`

Reply:
@mjsax Thanks so much! This works great. Much better than submitting a KIP to the Apache Kafka Debate Club :smile: 

CC @eduwercamacaro let's refactor this tomorrow.